### DISCUSSION TITLE
How to upload file and link it to Model?

### DISCUSSION THREAD

**Discussion by alexdieudonne** [directus/directus#10631] [Archive]
I can't upload an image through API and i would like to achieve this.
Like strapi
![119806d61f6aabfd7662ea1d62b44e6bfaddf4b1](https://user-images.githubusercontent.com/59718717/147008474-1aa63a95-cd19-427d-95ee-f315007cb307.png)

Thanks for directus it's incredible.

**Comment by rijkvanzanten** [comment]
Heya! You can't upload a file directly into another collection. You can upload files through the [Upload a File](https://docs.directus.io/reference/files/#upload-a-file) endpoint, and then link the UUID to the articles item you're creating 👍🏻

**Reply by auyongjinyoo** [reply]
May I know any reference how to link the UUID to the articles item through API?

**Reply by paescuj** [reply]
You should be able to go with something like this:
```js
// the ID of your article
const articleId = 1

// create the file
const formData = new FormData();
formData.append('title', 'My First File');
formData.append('file', fileInput.files[0]);
const file = await directus.files.createOne(formData);

// if you want to create a new item in "articles"
await directus.items('articles').createOne({
  id: articleId,
  // for singe file field
  file: file.id,
  // for multiple files field
  file: [file.id],
});

// if you want to update an existing item in "articles"
await directus.items('articles').updateOne(articleId, {
  file: [file.id],
});
```
